### PR TITLE
Swap plugin

### PR DESCRIPTION
**Pull request by Art17**
## Swap plugin
Provide functionality to swap ERC20 REM to REM.

### CODE REVIEW

File: plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+constexpr int32_t block_interval_ms = 500;
+constexpr int64_t block_timestamp_epoch = 946684800000ll;  // epoch is year 2000
+typedef eosio::chain::block_timestamp<block_interval_ms, block_timestamp_epoch> epoch_block_timestamp;
+
+const char* rem_token_id              = "REM";
+const char* eth_swap_contract_address = "0x9fB8A18fF402680b47387AE0F4e38229EC64f098";
+const char* eth_swap_request_event    = "0x0e918020302bf93eb479360905c1535ba1dbc8aeb6d20eff433206bf4c514e13";
+const char* return_chain_id           = "ethropsten";
+const char* chain_id                  = "93ece941df27a5787a405383a66a7c26d04e80182adf504365710331ac0625a7";
```
Comment (roman-tik) [inline plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp]: let's make it possible to override these params in the config

File: plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+const size_t request_swap_hex_data_length = 512;
```
Comment (roman-tik) [inline plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp]: Please describe with comments each constant, some of them will also need to be able to get overridden in the config

**Review by roman-tik — CHANGES_REQUESTED** [review]
Please create the eth_mainnet plugin as well and make them use common codebase, so we do not have the same code copies (e.g. my_web3, tests, etc) in these two plugins.

File: plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp (hunk 1 above)
Comment (Art17) [reply plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp]: Done.

File: plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp (hunk 2 above)
Comment (Art17) [reply plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp]: Done.

**Pull request merged**